frontend/src/metabase/writeback/components/ActionCreator/InlineDataReference.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react";
+import { t } from "ttag";
+
+import Icon from "metabase/components/Icon";
+
+import DataReference from "metabase/query_builder/components/dataref/DataReference";
+
+import {
+  DataReferenceContainer,
+  TriggerButton,
+} from "./InlineDataReference.styled";
+
+export const DataReferenceInline = ({
+  onClose,
+  isOpen,
+}: {
+  onClose: () => void;
+  isOpen: boolean;
+}) => {
+  const [dataRefStack, setDataRefStack] = useState<any[]>([]);
+
+  const pushRefStack = (ref: any) => {
+    setDataRefStack([...dataRefStack, ref]);
+  };
+
+  const popRefStack = () => {
+    setDataRefStack(dataRefStack.slice(0, -1));
+  };
+
+  return (
+    <DataReferenceContainer isOpen={isOpen}>
+      <DataReference
+        dataReferenceStack={dataRefStack}
+        popDataReferenceStack={popRefStack}
+        pushDataReferenceStack={pushRefStack}
+        onClose={onClose}
+      />
+    </DataReferenceContainer>
+  );
+};
+
+export const DataReferenceTriggerButton = ({
+  onClick,
+}: {
+  onClick: () => void;
+}) => (
+  <TriggerButton
+    title={t`Data Reference`}
```

Comment:
Does `title` actually do anything here? We seem to just display an icon. Maybe you were about to add a `tooltip`? Seems handy it this case